utils/yoke/src/yoke.rs

```diff
@@ -77,7 +78,7 @@ use alloc::sync::Arc;
 pub struct Yoke<Y: for<'a> Yokeable<'a>, C> {
     // must be the first field for drop order
     // this will have a 'static lifetime parameter, that parameter is a lie
-    yokeable: Y,
+    yokeable: KindaSortaDangling<Y>,
     cart: C,
```

Comment:
If you also wrap the cart in `KindaSortaDangling`, that should take care of https://github.com/unicode-org/icu4x/issues/2095, no?

Reply:
I was wondering about that. I wasn't sure if it still works with [`backing_cart()`](https://docs.rs/yoke/latest/yoke/struct.Yoke.html#method.backing_cart) which lets you materialize a `&Cart` reference.


I'm going t ocontinue this discussion in #2095 but yeah I think we should revisit this now, thanks for flagging!

Reply:
> I was wondering about that. I wasn't sure if it still works with [backing_cart()](https://docs.rs/yoke/latest/yoke/struct.Yoke.html#method.backing_cart) which lets you materialize a &Cart reference.

I think it's fine; `&Box<T>` only has an outer noalias for the shared ref and there is no way to get to the inner box.

Reply:
cool, in that case I think I'll solve this in a separate PR but I'll try and get to it!